seaworthy/dockerhelper.py

```diff
@@ -116,23 +155,41 @@ def create_container(self, name, image, network=None, **kwargs):
             namespace.
         :param image:
             The image tag or image object to create the container from.
-        :param docker.models.networks.Network network:
+        :param network:
             The network to connect the container to. The container will be
             given an alias with the ``name`` parameter. Note that, unlike the
-            Docker Python client, this parameter should be a ``Network`` model
-            object, and not just a network ID.
+            Docker Python client, this parameter can be a ``Network`` model
+            object, and not just a network ID or name.
+        :param volumes:
+            A mapping of volumes to bind parameters. Note that, unlike the
+            Docker Python client, the volumes in the mapping can be specified
+            using ``Volume`` model objects, and not just volume names.
```

Comment:
Are the model objects hashable? If you get two different model objects for the same entity, do they hash to the same value?

What happens if you have the same volume specified using both id and model object?

Reply:
Yes.. in `docker.models.resource.Model` (the base class of all model types) there is:
```python
def __eq__(self, other):
        return isinstance(other, self.__class__) and self.id == other.id

def __hash__(self):
        return hash("%s:%s" % (self.__class__.__name__, self.id))
```
so two models of the same type with the same ID should hash to the same thing.

When there is a model and ID string in the volumes hash provided, because we use a dict comprehension, the element that is iterated over last is the one that gets through 😕, which, depending on the Python version, could be predictable or not. Perhaps I should disallow this?

Reply:
What's the use case for passing model objects?

I don't think the current behaviour is really an issue -- I can't think of any cases in which it's likely that you'd end up with a mix of types in the dict.

Reply:
The use case is just consistency. The Docker Python client sometimes accepts both models and ID strings for things, but sometimes only ID strings (like for volumes). I'd like to encourage just using the models as much as possible.